### PR TITLE
Add package level compilation to ProjectAPI

### PR DESCRIPTION
**Pull request by Kishanthan**
## Purpose
$subject.

Related to #26262

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/build/ModuleCompilation.java [hunk 1]
```diff
@@ -66,9 +66,21 @@
         return new DependencyGraph<>(dependencyIdMap);
     }
 
-    void addModuleDependencies(ModuleId moduleId, Map<ModuleId, Set<ModuleId>> dependencyIdMap) {
+    private void addModuleDependencies(ModuleId moduleId, Map<ModuleId, Set<ModuleId>> dependencyIdMap) {
         Package pkg = packageResolver.getPackage(moduleId.packageId());
         Collection<ModuleId> directDependencies = pkg.moduleDependencyGraph().getDirectDependencies(moduleId);
+
+        ModuleContext moduleContext = pkg.packageContext().moduleContext(moduleId);
```
Comment (pramodya1994) [inline compiler/ballerina-lang/src/main/java/io/ballerina/build/ModuleCompilation.java:73]: `moduleContext` hides the class level field defined in L41. Can we use other name for this?

Comment (Kishanthan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/build/ModuleCompilation.java]: I was on it.

File: project-api/ballerina-projects/src/main/java/io/ballerina/projects/env/CustomPackageResolver.java [hunk 2]
```diff
@@ -65,7 +65,8 @@ public CustomPackageResolver(Project project) {
                 // TODO load the module from the custom_project_dir
                 module = loadFromCustomProjectDir(modLoadRequest);
             }
-            modLoadResponses.add(new ModuleLoadResponse(currentPkg.packageId(), module.moduleId(), modLoadRequest));
+            modLoadResponses.add(new ModuleLoadResponse(module.packageInstance().packageId(), module.moduleId(),
```
Comment (Kishanthan) [inline project-api/ballerina-projects/src/main/java/io/ballerina/projects/env/CustomPackageResolver.java:68]: @sameerajayasoma , using `currentPkg.packageId()` seems to be a bug here which was the issue for package dependencies becoming empty (which we saw during the debug session)

Comment (pramodya1994) [reply project-api/ballerina-projects/src/main/java/io/ballerina/projects/env/CustomPackageResolver.java:68]: I think we should get `packageName` from the `Ballerina.toml` file. At some places, we are taking the project directory name as the default package name. I think this creates errors.

Comment (sameerajayasoma) [reply project-api/ballerina-projects/src/main/java/io/ballerina/projects/env/CustomPackageResolver.java:68]: AFAIR, here we are only considering the `PackageId`.

**Review by sameerajayasoma — APPROVED** [review]

**Pull request merged**